pymongo/pool.py

```diff
@@ -552,7 +552,8 @@ def unpin(self):
     def hello_cmd(self):
         # Handshake spec requires us to use OP_MSG+hello command for the
         # initial handshake in load balanced or versioned api mode.
-        if self.opts.server_api or self.hello_ok or self.opts.load_balanced:
+        if (self.opts.server_api or self.hello_ok or self.opts.load_balanced
+            or self.max_wire_version >= 6):
```

Comment:
This change isn't needed. We already set op_msg_enabled to true when `max_wire_version >= 6)` elsewhere.

Reply:
Oops, yeah. You're correct. This was due to me confusing [DRIVERS-1916](https://jira.mongodb.org/browse/DRIVERS-1916) with saying that OP_MSG necessitates "hello" not the other way around. Reverted.